php/commands/sites.php

```diff
@@ -228,15 +229,28 @@ public function aliases($args, $assoc_args) {
     $print = Input::optional('print', $assoc_args, false);
     $json = \Terminus::get_config('json');
     $location = Input::optional('location', $assoc_args, getenv("HOME").'/.drush/pantheon.aliases.drushrc.php');
-    $message = "Pantheon aliases updated.";
-    if (!file_exists($location)) {
-      $message = "Pantheon aliases created.";
+
+    // Cannot provide just a directory
+    if (is_dir($location)) {
+      \Terminus::error("Please provide a full path with filename, e.g. %s/pantheon.aliases.drushrc.php", $location);
+      exit(1);
+    }
+
+    $file_exists = file_exists($location);
+
+    // Create the directory if it doesn't yet exist
+    $dirname = dirname($location);
+    if (!is_dir($dirname)) {
+      mkdir($dirname, 0755, true);
     }
+
     $content = $user->getAliases();
     $h = fopen($location, 'w+');
     fwrite($h, $content);
     fclose($h);
-    chmod($location, 0777);
+    chmod($location, 0755);
```

Comment:
@greg-1-anderson thanks for the feedback. I think I will go with `0700` (a reasonable default) and expect their umask to make it more permissible if that's their setting.